src/test/munit/aggregators-scenarios-test-case.xml

```diff
@@ -181,6 +181,63 @@
         <munit-tools:queue/>
     </flow>
 
+    <munit:test name="testTimeBasedAggregationInsideScatterGather"
```

Comment:
I would add a test with several events sent to a flow to verify that in that case a group-based aggregation does not override events. Is that covered in the added test in some way?

Reply:
Hi! I added the test (testGroupBasedAggregationInsideForeach) that checks that group-based aggregator's events are not overridden. And I made some changes to the code to make it backward compatible using kryo deserialization in the last commit.

Reply:
Great!